src/cpu.c

```diff
@@ -70,7 +70,8 @@ unsigned int indirectYAddressMode(struct CPU *cpu, struct BUS *bus) {
 }
 
 unsigned int relativeAddressMode(struct CPU *cpu, struct BUS *bus) {
-  signed char address = cpu->pc + readBus(cpu->pc, bus);
+  // this cast should stay here cause readBus return an unsigned byte so the math fails when for negative representations
```

Comment:
I think we can remove the comment,  the function signature should remain as unsigned and the return ad signed ? Can't we make then equal?

Reply:
Well, after our conversation about the addressing, all this methods should return addresses only, so, this means only unsigned stuff.
I'll try to make them equally unsigned and run tests

Reply:
👍